inject/android.js

```diff
@@ -16,10 +25,43 @@ function initCode () {
         platform: processedConfig.platform
     })
 
+    const messageSecret = processedConfig.messageSecret
+    const messageCallback = processedConfig.messageCallback
+    const messageInterface = processedConfig.messageInterface
+
+    const wrappedUpdate = ((providedSecret, ...args) => {
+        if (providedSecret === messageSecret) {
+            update(...args)
+        }
+    }).bind()
+
+    Object.defineProperty(window, messageCallback, {
+        value: wrappedUpdate
+    })
+
     init(processedConfig)
 
-    // Not supported:
-    // update(message)
+    window.addEventListener('sendMessageProxy' + messageSecret, event => {
+        event.stopImmediatePropagation()
+
+        if (!(event instanceof CustomEvent) || !event?.detail) {
+            return console.warn('no details in sendMessage proxy', event)
+        }
+
+        const messageType = event.detail?.messageType
+        if (!allowedMessages.includes(messageType)) {
```

Comment:
Note: It could be worth checking the message is allowed in the Android code as well.

Reply:
I've made a task to remind me to do that 👍